tests/test_job.py

```diff
@@ -943,7 +972,82 @@ def test_dependents_are_met_if_dependency_is_deleted(self):
 
         assert dependent_job.dependencies_are_met()
         assert dependent_job.get_status() == JobStatus.QUEUED
-    
+
+    def test_rpush_fixture(self):
+        fixtures.rpush('foo', 'bar')
+        assert as_text(self.testconn.lrange('foo', 0, 0)[0]) == 'bar'
+
+    def test_execution_order_with_sole_dependency(self):
+        queue = Queue(connection=self.testconn)
+        key = 'test_job:job_order'
+        w1 = Worker([queue])
+        w2 = Worker([queue])
+
+        # When there are no dependencies, the two fast jobs ("A" and "B") run in the order enqueued.
+        queue.enqueue_call(fixtures.long_running_job, kwargs={'timeout': 0.5}, job_id='slow_guy')
+        queue.enqueue_call(fixtures.rpush, args=[key, "A"])
+        queue.enqueue_call(fixtures.rpush, args=[key, "B"])
+
+        self.testconn.delete(key)
+        w1.work(burst=True)
+        w2.work(burst=True)
```

Comment:
Why do we need to use two workers here? It seems like `w1` would just be running all three jobs serially.

Reply:
I think you're right, and I wanted to ask advice here.

The fact that these unit tests actually pass, given that in reality only one worker is
doing all the jobs, is actually quite interesting. I guess it demonstrates that jobs with
dependencies get moved to the deferred job registry at the time they're enqueued,
and that this affects execution order.

But I don't really feel it proves what these unit tests _want_ to prove, i.e., that in
a situation with multiple workers, the free workers really will skip over dependent jobs
until those jobs are ready to go.

If we can get two workers going, we can test that. Is there an easy way to do that,
within a unit test? I tried messing around with `os.fork()`, but it's failing, and I
don't understand why. (Maybe because it's all happening within `pytest`? I can post some
code if you want.)

Reply:
I'd rather not mess around with `os.fork()` during testing - it feels somewhat hackish.

> But I don't really feel it proves what these unit tests want to prove, i.e., that in
a situation with multiple workers, the free workers really will skip over dependent jobs
until those jobs are ready to go.
We can do this by comprehensively testing `job.dependencies_are_met()` - this method should only return `True` if all dependencies have been satisfied.

Reply:
Agreed, `os.fork()` is probably best avoided. : ) However I found some existing unit tests in `test_worker.py` and `test_commands.py` where you use `multiprocessing` to achieve something similar, which gave me some ideas. So I've tried again, and I think these tests are in good shape now.

I'm not sure if I understood what you meant about using `job.dependencies_are_met()`, but I've made an attempt in unit test `test_dependencies_are_met_at_execution_time`.